crates/matrix-sdk/src/test_utils/mocks.rs

```diff
@@ -0,0 +1,389 @@
+            server: &self.server,
+            sync_response_builder: self.sync_response_builder.clone(),
+        }
+    }
+
+    /// Creates a prebuilt mock for sending an event in a room.
+    ///
+    /// Note: works with *any* room.
+    pub fn mock_room_send(&self) -> MockRoomSend<'_> {
+        let mock = Mock::given(method("PUT"))
+            .and(path_regex(r"^/_matrix/client/r0/rooms/.*/send/.*"))
+            .and(header("authorization", "Bearer 1234"));
+        MockRoomSend { mock, server: &self.server }
+    }
+
+    /// Creates a prebuilt mock for asking whether *a* room is encrypted or not.
+    ///
+    /// Note: Applies to all rooms.
+    pub fn mock_room_state_encryption(&self) -> MockEncryptionState<'_> {
+        let mock = Mock::given(method("GET"))
+            .and(header("authorization", "Bearer 1234"))
+            .and(path_regex(r"^/_matrix/client/r0/rooms/.*/state/m.*room.*encryption.?"));
+        MockEncryptionState { mock, server: &self.server }
+    }
+
+    pub fn mock_set_room_state_encryption(&self) -> MockSetEncryptionState<'_> {
+        let mock = Mock::given(method("PUT"))
+            .and(header("authorization", "Bearer 1234"))
+            .and(path_regex(r"^/_matrix/client/r0/rooms/.*/state/m.*room.*encryption.?"));
+        MockSetEncryptionState { mock, server: &self.server }
+    }
+
+    /// Creates a prebuilt mock for the room redact endpoint.
+    pub fn mock_room_redact(&self) -> MockRoomRedact<'_> {
+        let mock = Mock::given(method("PUT"))
+            .and(path_regex(r"^/_matrix/client/r0/rooms/.*/redact/.*?/.*?"))
+            .and(header("authorization", "Bearer 1234"));
+        MockRoomRedact { mock, server: &self.server }
+    }
+
+    /// Creates a prebuilt mock for retrieving an event with /room/.../event.
+    pub fn mock_room_event(&self) -> MockRoomEvent<'_> {
+        let mock = Mock::given(method("GET")).and(header("authorization", "Bearer 1234"));
+        MockRoomEvent { mock, server: &self.server, room: None, match_event_id: false }
+    }
+}
+
+/// Parameter to [`MatrixMockServer::sync_room`].
+pub enum AnyRoomBuilder {
+    Invited(InvitedRoomBuilder),
+    Joined(JoinedRoomBuilder),
+    Left(LeftRoomBuilder),
+    Knocked(KnockedRoomBuilder),
+}
+
+impl From<InvitedRoomBuilder> for AnyRoomBuilder {
+    fn from(val: InvitedRoomBuilder) -> AnyRoomBuilder {
+        AnyRoomBuilder::Invited(val)
+    }
+}
+
+impl From<JoinedRoomBuilder> for AnyRoomBuilder {
+    fn from(val: JoinedRoomBuilder) -> AnyRoomBuilder {
+        AnyRoomBuilder::Joined(val)
+    }
+}
+
+impl From<LeftRoomBuilder> for AnyRoomBuilder {
+    fn from(val: LeftRoomBuilder) -> AnyRoomBuilder {
+        AnyRoomBuilder::Left(val)
+    }
+}
+
+impl From<KnockedRoomBuilder> for AnyRoomBuilder {
+    fn from(val: KnockedRoomBuilder) -> AnyRoomBuilder {
+        AnyRoomBuilder::Knocked(val)
+    }
+}
+
+/// A wiremock Mock wrapping also the server, to not have to pass it around when
+/// mounting.
```

Comment:
I don't understand this. Can you find a different way to write it? I think maybe it's because I don't know much about wiremock.